internal/featuretests/featuretests.go

```diff
@@ -85,11 +86,11 @@ func setup(t *testing.T, opts ...interface{}) (cache.ResourceEventHandler, *Cont
 
 	rand.Seed(time.Now().Unix())
 
-	statusCache := &k8s.StatusCacher{}
-
+	statusCache := status.NewCache()
+	statusUpdateCacher := &k8s.StatusUpdateCacher{}
```

Comment:
Once you flip this, you can delete the `StatusClient` interface and the `StatusCache` implementation.

Maybe `StatusUpdateCacher` can take the `StatusCache` name or move into the `fixtures` package.